### PR TITLE
Add Reusable Statistics Chart Components

### LINKED ISSUE #289: Add Reusable Statistics Chart Components
**Issue:** Add Reusable Statistics Chart Components

Create reusable, animated chart components for visualizing event statistics. This includes a horizontal bar chart for campus distribution, a donut chart for age distribution, a line chart for registration timeline, circular percentage indicators, animated counters, and chart legends. All components should support smooth animations and follow the app's design system.

### PR DESCRIPTION
**Pull request by rita-naimi**
## What?

Add reusable, animated chart components for visualizing event statistics data. This PR introduces six composable components:
- `AnimatedHorizontalBarChart` - For displaying campus distribution
- `AnimatedDonutChart` - For displaying age distribution  
- `AnimatedLineChart` - For displaying registration timeline
- `CircularPercentageIndicator` - For displaying attendees/followers rate
- `AnimatedCounter` - For animated number displays
- `ChartLegend` - For chart color legends

## Why?

These chart components are needed to build the Event Statistics screen (which will be the subject of another PR), which will allow organizers to view insights about their events. By extracting charts into reusable components:
- They can be reused across the app for other potential analytics features
- Each component can be tested in isolation
- The main screen code remains clean and focused on layout

## How?

- Created `StatisticsConstants.kt` with all UI dimensions, animation durations, and numeric values to avoid magic numbers
- Added test tags to `C.kt` for all chart components (`stats_bar_chart`, `stats_donut_chart`, etc.)
- Implemented charts using Jetpack Compose `Canvas` with custom drawing logic
- Used `Animatable` and `animateFloatAsState` for smooth entry animations
- Applied `MaterialTheme.typography` and `MaterialTheme.colorScheme` for consistent styling
- Used `stringResource()` for accessibility content descriptions
- Added 17 comprehensive UI tests covering all components and edge cases

Fixes #289

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsConstants.kt
Comment (haaissa) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsConstants.kt]: Incredible work!

File: app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+ * Animated line chart for displaying temporal data.
+ *
+ * @param data List of data points with timestamps.
+ * @param lineColor Color of the line.
+ * @param fillColor Color for the gradient fill under the line.
+ * @param animationProgress Progress of the animation (0f to 1f).
+ * @param modifier Modifier for the chart container.
+ */
+@Composable
+fun AnimatedLineChart(
```
[the remark this answers is not in the document]
Comment (rita-naimi) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt]: Yes, done! 👍

File: app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+              val radius = (size.minDimension - strokeWidth) / 2
+              val center = Offset(size.width / 2, size.height / 2)
+
+              var startAngle = StatisticsConstants.CHART_START_ANGLE
+              val totalPercentage = segments.sumOf { it.second.toDouble() }.toFloat()
+              val sweepMultiplier = animatedSweep.value
+
+              segments.forEachIndexed { index, (_, percentage) ->
+                val sweepAngle =
+                    (percentage / totalPercentage) *
```
[the remark this answers is not in the document]
Comment (rita-naimi) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt]: Good catch! 👀 But the division is actually protected, I extracted this into a helper function `calculateDonutSweepAngle` which has a guard on line 129: `if (totalPercentage == 0f) return 0f`. This prevents division by zero when segments are empty or all have zero values. The test `calculateDonutSweepAngle_allBranches` explicitly covers this case!

File: app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+              .fillMaxWidth()
+              .height(StatisticsConstants.LINE_CHART_HEIGHT)
+              .testTag(C.Tag.stats_line_chart)
+              .semantics { contentDescription = chartDescription }) {
+        val paddingBottom = StatisticsConstants.LINE_CHART_PADDING_BOTTOM.toPx()
+        val paddingTop = StatisticsConstants.LINE_CHART_PADDING_TOP.toPx()
+        val chartHeight = size.height - paddingBottom - paddingTop
+        val chartWidth = size.width
+
+        val points = calculateChartPoints(data, chartWidth, chartHeight, paddingTop)
```
[the remark this answers is not in the document]
Comment (haaissa) [reply app/src/main/java/com/github/se/studentconnect/ui/screen/statistics/StatisticsCharts.kt]: We just need to tell Amir since he’ll be the one doing this task (maybe he’s already started, so the sooner the better).
